utils/mixins/apiUrlMixin.ts

```diff
@@ -1,17 +1,20 @@
-import { Component, Vue } from 'nuxt-property-decorator'
+import { Component, mixins } from 'nuxt-property-decorator'
 import { getChainEndpointByPrefix } from '@/utils/chain'
+import PrefixMixin from '@/utils/mixins/prefixMixin'
+import { DEFAULT_PREFIX } from '@kodadot1/static'
 
 /*
  * refer to https://stackoverflow.com/questions/51873087/unable-to-use-mixins-in-vue-with-typescript
  * import { Component, Mixins } from 'nuxt-property-decorator';
  * class ExtendedClass extends Mixins(ActualMixin) {
  */
 @Component
-export default class ApiUrlMixin extends Vue {
+export default class ApiUrlMixin extends mixins(PrefixMixin) {
   get apiUrl() {
-    const endpoint = getChainEndpointByPrefix(
-      this.$store.getters.currentUrlPrefix
-    )
-    return endpoint || this.$store.getters.getSettings['apiUrl']
+    const endpoint: string =
+      getChainEndpointByPrefix(this.urlPrefix) ||
```

Comment:
this should be enough, otherwise we it should throw err.
but can be done in different scope

Reply:
I don't understand your comment but ok

Reply:
That it should throw error if we cannot find url with prefix